test/unit/repo_test.rb

```diff
@@ -126,4 +126,53 @@ class RepoTest < ActiveSupport::TestCase
     assert_not repos.include?(subscribed_repo)
     assert repos.include?(unsubscribed_repo)
   end
+
+  test 'weight and color' do
+    @repo = Repo.create user_name: 'Blofeld', name: 'world_domination', issues_count: -1
+    @repo = Repo.new(user_name: 'Blofeld',
+                     name: 'world_domination',
+                     issues_count: -1)
+    assert_equal @repo.weight, ''
+    assert_equal @repo.color, 'lightgrey'
+
+    @repo.issues_count = nil
+    assert_equal @repo.weight, ''
+    assert_equal @repo.color, 'lightgrey'
+
+    @repo.issues_count = 0
+    assert_equal @repo.weight, 'low'
+    assert_equal @repo.color, '5bb878'
+
+    @repo.issues_count = 100
+    assert_equal @repo.weight, 'low'
+    assert_equal @repo.color, '5bb878'
```

Comment:
I think for this case, creating tests for the minimum & maximum value for each weight category threshold will represent the function clearly enough. I would suggest only keeping the test for:
-1, 0, 199, 200, 799, 800